src/dra/72BB0.c

```diff
@@ -624,8 +624,398 @@ void func_80113F7C(void) {
+            sfxIndex = (rand() & 1) + 3;
+            switch (arg1) {
+            case 0:
+            case 1:
+                break;
+            case 3:
+            case 4:
+                var_s0 = FIX(1);
+                break;
+            }
+            PLAYER.velocityY = var_s0 + FIX(-4);
+            func_8010E3B8(FIX(-1.66666));
+            PLAYER.step_s = 1;
+            if (func_80113E68() == 0) {
+                PLAYER.ext.generic.unkAC = 0x40;
+            }
+            break;
+        case 3:
+        case 7:
+            sfxIndex = (rand() & 1) + 3;
+            switch (arg1) {
+            case 3:
+            case 4:
+                var_s0 = FIX(1);
+            case 0:
+            default:
+                PLAYER.velocityY = var_s0 + FIX(-4);
+                func_8010E3B8(FIX(-1.66666));
+                ;
+                PLAYER.step_s = 1;
+                if (func_80113E68() == 0) {
+                    PLAYER.ext.generic.unkAC = 0x40;
+                }
+                break;
+            case 2:
+            case 6:
+                PLAYER.velocityY = 0;
+                func_8010E3B8(FIX(-2.5));
+                PLAYER.step_s = 7;
+                PLAYER.ext.generic.unkAC = 0x23;
+                CreateEntFactoryFromEntity(g_CurrentEntity, 0, 0);
+            }
+            break;
+        case 2:
+            sfxIndex = (rand() & 1) + 5;
+            switch (arg1) {
+            case 0:
+            case 1:
+                PLAYER.velocityY = 0;
+                func_8010E3B8(FIX(-2.5));
+                PLAYER.step_s = 6;
+                randbit = (rand() & 1);
+                // Weird test, I  think PLAYER.entityRoomIndex is a wrong name.
+                // But if they don't match, we add 2 to the offset.
+                unkAC_offset =
+                    randbit |
+                    ((PLAYER.entityRoomIndex != PLAYER.facingLeft) ? 2 : 0);
+                PLAYER.ext.generic.unkAC = unkAC_offset + 0x31;
+                CreateEntFactoryFromEntity(
+                    g_CurrentEntity, FACTORY(0x600, 0), 0);
+                break;
+            case 2:
+            case 6:
+                PLAYER.velocityY = 0;
+                func_8010E3B8(FIX(-2.5));
+                PLAYER.step_s = 7;
+                PLAYER.ext.generic.unkAC = 0x23;
+                CreateEntFactoryFromEntity(g_CurrentEntity, FACTORY(0, 0), 0);
+                break;
+            case 3:
+            case 4:
+            case 5:
+            default:
+                PLAYER.velocityY = FIX(-2);
+                func_8010E3B8(FIX(-1.25));
+                PLAYER.step_s = 1;
+                func_80113E68();
+                break;
+            }
+            break;
+        default:
+            FntPrint("dam_kind:%04x\n", arg0->damageKind);
+            break;
+        }
+        g_Player.unk40 = 0x8166;
+        g_Player.D_80072F00[2] = 6;
+        PlaySfx(D_800ACF84[sfxIndex]);
+        if (arg0->unk0 & 0x100) {
```

Comment:
Decided to name `unk0` as `effects` since I think it is also responsible for things like attacks which have an Ice element attached. Future analysis will probably be able to work this out. I identified many of these elements in my `assets.py` system of extracting the equipments and they may have the same bit order here.